x-pack/heartbeat/monitors/browser/synthexec/synthcmd.go

```diff
@@ -0,0 +1,43 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+//go:build linux || darwin || synthetics
+
+package synthexec
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+// Variant of exec.command with redacted params, which could contain sensitive stuff
+type SynthCmd struct {
+	*exec.Cmd
+}
+
+func (cmd *SynthCmd) String() string {
+	b := new(strings.Builder)
+	b.WriteString(cmd.Path)
+	for i := 1; i < len(cmd.Args); i++ {
```

Comment:
why not use range?

Reply:
There's lookahead involved, we want to redact the parameter value following the key

Reply:
Ahh I see, missed that.